centreon/packages/ui/src/Graph/BasicComponents/Lines/index.tsx

```diff
@@ -0,0 +1,171 @@
+    leftScale,
+    rightScale
+  });
+
+  const commonStackedLinesProps = {
+    displayAnchor,
+    graphHeight: height,
+    graphSvgRef,
+    graphWidth: width,
+    xScale,
+    yScale: stackedYScale
+  };
+
+  return (
+    <g>
+      {(areaStackedLines?.display ?? true) && (
+        <>
+          {displayArea(stackedLinesData.lines) && (
+            <StackedLines
+              lines={stackedLinesData.lines}
+              timeSeries={stackedLinesData.timeSeries}
+              {...commonStackedLinesProps}
+            />
+          )}
+
+          {displayArea(invertedStackedLinesData.lines) && (
+            <StackedLines
+              lines={invertedStackedLinesData.lines}
+              timeSeries={invertedStackedLinesData.timeSeries}
+              {...commonStackedLinesProps}
+            />
+          )}
+        </>
+      )}
+
+      {displayThresholdArea && (
+        <WrapperThresholdLines
+          areaThresholdLines={areaThresholdLines}
+          graphHeight={height}
+          leftScale={leftScale}
+          lines={displayedLines}
+          rightScale={rightScale}
+          timeSeries={timeSeries}
+          xScale={xScale}
+        />
+      )}
+
+      {displayAreaRegularLines
+        ? regularLines.map(
+            ({
+              metric,
+              areaColor,
+              transparency,
+              lineColor,
+              filled,
+              unit,
+              highlight,
+              invert
+            }) => {
+              const [, secondUnit, thirdUnit] = getUnits(
+                regularLines as Array<Line>
+              );
+              const yScale = getYScale({
+                hasMoreThanTwoUnits: !isNil(thirdUnit),
+                invert,
+                leftScale,
+                rightScale,
+                secondUnit,
+                unit
+              });
+
+              return (
+                <g key={metric}>
+                  {displayAnchor && (
+                    <RegularAnchorPoint
+                      areaColor={areaColor}
+                      graphHeight={height}
+                      graphWidth={width}
+                      lineColor={lineColor}
+                      metric={metric}
+                      timeSeries={timeSeries}
+                      transparency={transparency}
+                      xScale={xScale}
+                      yScale={yScale}
+                    />
+                  )}
+                  <RegularLine
+                    areaColor={areaColor}
+                    filled={filled}
+                    graphHeight={height}
+                    highlight={highlight}
+                    lineColor={lineColor}
+                    metric={metric}
+                    timeSeries={timeSeries}
+                    transparency={transparency}
+                    unit={unit}
+                    xScale={xScale}
+                    yScale={yScale}
+                  />
+                </g>
+              );
```

Comment:
by adding the `AnchorPoint` before the actual line it is beneath the line visually,
on one hand it could be simply swapped

only i notice that this approach has some issues :
because `AnchorPoint` also contains the guiding lines, it draws them for each `Line` instance / value... so there's always depth issues with the other lines, and draws above the other dots / points

structure wise, there are multiple layers / depths :
1. bottom layer: background grid
2. data layer: the values / `Lines`
3. axis layer: overlaps the border of your data layer, keeps it tidy visually
4. info/interaction layer: the dots / "anchor points" and guiding lines + tooltips 

wdyt? feasible to restructure to avoid the depth issues ?

Reply:
info/interaction layer: the dots / "anchor points" and guiding lines + tooltips **+ click + select the time period with the mouse**

Reply:
after discussion via slack :
for the layers:
1- the basic components are displayed first (lines, grids , axis ) - first layer -
2- depends on the intercation of the user we display an extra layer (anchorPoint (dots, guidin lines)
--> this is the current behavior so i didn t see an issue with layers 👀 
-the issue of guiding lines there is a redundant wich is solved , and the dots should be displayed first (the fix is i create a didicate component for guidingLines because it s depends just on mousePosition data not the lines of regularLines ... )

![fixAncgor](https://github.com/centreon/centreon/assets/97687698/4afa2efe-2271-42e7-9f1f-22daf33b8092)